payjoin-cli/tests/e2e.rs

```diff
@@ -457,7 +457,14 @@ mod e2e {
             let db = docker.run(Redis::default());
             let db_host = format!("127.0.0.1:{}", db.get_host_port_ipv4(6379));
             println!("Database running on {}", db.get_host_port_ipv4(6379));
-            payjoin_directory::listen_tcp_with_tls(port, db_host, timeout, local_cert_key).await
+            payjoin_directory::listen_tcp_with_tls(
+                format!("http://localhost:{}", port),
+                port,
```

Comment:
I'm not sure I follow why `port` needs to be specified in the base_url _and_ as a separate parameter. Is it just a testing/docker thing?

Reply:
In prod, you would just pass e.g. "https://payjo.in" where the port can be assumed by the scheme (http/https), but in testing, you need to pass the explicit port because it's non-default. The base_url could be a totally separate port to the one exposed by the server if it were behind a reverse proxy or something.